File: Backend/Online-Store-Backend/Online-Store-Backend.Database/Categories/Models/Property.cs
using Online_Store_Backend.Core.Entity;
using Online_Store_Backend.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;

namespace Online_Store_Backend.Database.Categories.Models
{
    [Serializable]
    [DataContract]
    [Table("category_prop", Schema = "public")]
    public class Property : BaseEntity
    {
        [DataMember]
        public Int64 CategoryID { get; set; }

        [ForeignKey("CategoryID")]
        public virtual Category? Category { get; set; }

        [DataMember]
        [Required]
        [MaxLength(100)]
        public String? PropName { get; set; }

        [DataMember]
        [Required]
        public PropValueType ValueType { get; set; }
    }
}

File: Frontend/Online-Store-App.CommonLib/src/Models/Data/Property.ts
import { type BaseEntity } from "./Base/BaseEntity";

export interface Property extends BaseEntity {
    categoryID: number;
    propName: string;
    valueType: number;
}

export interface PropertyForPost {
    categoryID: number;
    propName: string;
    valueType: number;
}

export interface PropertyForPut {
    id: number
    categoryID: number;
    propName: string;
    valueType: number;
}
